### PR TITLE
Fix a bug of AUTH_FAILURE in RouterServerTestFramework

### PR DESCRIPTION
**Pull request by zzmao**
Need to override callback onCompletion to continue the chain. 

Need help from @cgtz to enhance the framework for the case of chain stop in the middle.

### CODE REVIEW

**Comment by vgkholla** [comment]
Did you verify that the chain continues past the auth failure check now?

File: ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerTestFramework.java [hunk 1]
```diff
@@ -358,7 +358,12 @@ void check() throws Exception {
    * @param opChain the {@link OperationChain} object that this operation is a part of.
    */
   private void startGetBlobAuthorizationFailTest(final OperationChain opChain) {
-    Callback<GetBlobResult> callback = new TestCallback<>(opChain, false);
+    Callback<GetBlobResult> callback = new TestCallback<GetBlobResult>(opChain, false) {
```
Comment (cgtz) [inline ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerTestFramework.java]: Are you expecting an error to occur here? If so, you could set `expectError` to true, which will let the chain continue

Comment (cgtz) [reply ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerTestFramework.java]: e.g. `Callback<GetBlobResult> callback = new TestCallback<GetBlobResult>(opChain, true);`

Comment (zzmao) [reply ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerTestFramework.java]: oh. sure.

**Comment by zzmao** [comment]
Addressed @cgtz 's comments. Also add assert to make sure operations.size() is 0 in the end of the tests.

If there is IOException in blobId creation:
```
java.io.IOException: exception......

	at com.github.ambry.server.RouterServerTestFramework.startGetBlobAuthorizationFailTest(RouterServerTestFramework.java:368)
	at com.github.ambry.server.RouterServerTestFramework.continueChain(RouterServerTestFramework.java:509)
	at com.github.ambry.server.RouterServerTestFramework.startAwaitCreation(RouterServerTestFramework.java:457)
	at com.github.ambry.server.RouterServerTestFramework.continueChain(RouterServerTestFramework.java:503)
	at com.github.ambry.server.RouterServerTestFramework.access$300(RouterServerTestFramework.java:65)

```

If error code AuthorizationFailure doesn't match: 
```

java.lang.AssertionError: Blob should have failed in operation: getBlobAuthorizationFail-failureExpected-0

	at org.junit.Assert.fail(Assert.java:91)
	at com.github.ambry.server.RouterServerTestFramework$TestFuture.checkExpectedRouterErrorCode(RouterServerTestFramework.java:664)
	at com.github.ambry.server.RouterServerTestFramework$6.check(RouterServerTestFramework.java:391)
	at com.github.ambry.server.RouterServerTestFramework.checkOperationChains(RouterServerTestFramework.java:133)
	
```
If chain stops in the middle:
```
java.lang.AssertionError: opChain stops in the middle. 
Expected :0
Actual   :8
 <Click to see difference>


	at org.junit.Assert.fail(Assert.java:91)
	at org.junit.Assert.failNotEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:126)
	at org.junit.Assert.assertEquals(Assert.java:470)
	at com.github.ambry.server.RouterServerTestFramework.checkOperationChains(RouterServerTestFramework.java:136)
	at com.github.ambry.server.RouterServerPlaintextTest.largeBlobTest(RouterServerPlaintextTest.java:246)```

File: ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerTestFramework.java [hunk 2]
```diff
@@ -133,6 +133,7 @@ void checkOperationChains(List<OperationChain> opChains) throws Exception {
           testFuture.check();
         }
       }
+      Assert.assertEquals("opChain stops in the middle.", 0, opChain.operations.size());
```
Comment (vgkholla) [inline ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerTestFramework.java]: minor: `opChain stopped in the middle`

**Pull request merged**